site/search/SearchChartHitSmall.tsx

```diff
@@ -0,0 +1,306 @@
+                entities: [entityForDisplay],
+            })
+            const url = constructChartInfoUrl({ hit, grapherParams })
+            if (!url) return null
+            return fetchJson<GrapherValuesJson>(url)
+        },
+        // Only fetch when the component is visible
+        enabled: hasBeenVisible,
+    })
+
+    // The first chart tab is the primary chart type
+    const chartType: GrapherChartType | undefined = hit.availableTabs.find(
+        (tab) =>
+            tab !== GRAPHER_TAB_NAMES.Table &&
+            tab !== GRAPHER_TAB_NAMES.WorldMap
+    )
+
+    const dataDisplayProps = buildDataDisplayProps({
+        chartInfo,
+        chartType,
+        entity: entityForDisplay,
+        isEntityPickedByUser: hasUserPickedEntities,
+    })
+
+    const grapherParams = toGrapherQueryParams({ entities })
+    const chartUrl = constructChartUrl({ hit, grapherParams })
+
+    return (
+        <article ref={ref} className="search-chart-hit-small">
+            <div className="search-chart-hit-small__content">
+                <a
+                    href={chartUrl}
+                    className="search-chart-hit-small__title-link"
+                    onClick={onClick}
+                    data-algolia-index={getIndexName(
+                        SearchIndexName.ExplorerViewsMdimViewsAndCharts
+                    )}
+                    data-algolia-object-id={hit.objectID}
+                    data-algolia-position={hit.__position}
+                >
+                    <header className="search-chart-hit-small__header">
+                        <div className="search-chart-hit-small__title-container">
+                            <h3 className="search-chart-hit-small__title">
+                                {hit.title}
+                            </h3>
+                            {chartInfo?.source && (
+                                <span className="search-chart-hit-small__source">
+                                    {chartInfo.source}
+                                </span>
+                            )}
+                        </div>
+                        <Highlight
+                            hit={hit}
+                            attribute="subtitle"
+                            highlightedTagName="strong"
+                            classNames={{
+                                root: "search-chart-hit-small__subtitle",
+                            }}
+                        />
+                    </header>
+                </a>
+                <div className="search-chart-hit-small__tabs-container">
+                    {hit.availableTabs.map((tab) => {
+                        // Single-time line charts automatically switch to a
+                        // discrete bar chart. For the line chart tab preview URL,
+                        // we need to set the time parameter that forces Grapher
+                        // to show an actual line chart instead of the bar chart.
```

Comment:
Could you explain more why we do this? Why should the behavior be different when you arrive at the link through search vs some other way?

Reply:
It’s different because we don’t usually advertise something as a ‘line chart’. But here we give users two options, either go to the line chart or the bar chart. Without this extra care both links would end up going to the bar chart.

Here’s an example chart: https://ourworldindata.org/grapher/human-development-index?tab=line

It's quote on quote 'correct' to show a bar chart here because the handles are both on the same time. Grapher _could_ be more opinionated and switch to a different time selection in this case, but I think that would be quite messy and I usually try not to programatically change authored settings.

Reply:
Huh, what a complex problem! This clears it up for me well:

> It’s different because we don’t usually advertise something as a ‘line chart’. But here we give users two options, either go to the line chart or the bar chart. Without this extra care both links would end up going to the bar chart.

Could you add that to the comment in the code, pls?

And this seems like a (unrelated?) bug to me:

<img width="640" height="541" alt="image" src="https://github.com/user-attachments/assets/a53f7d9b-d7ac-4ed3-b648-d5dff6ed43ad" />

Shouldn't we also change the query param?